test/ui/cypress/support/auth.js

```diff
@@ -0,0 +1,51 @@
+import { verifyRequiredEnvVariables } from './../e2e/environment_helpers'
+
+Cypress.Commands.add('loginWithRHAccount', () => {
+   verifyRequiredEnvVariables()
+
+    cy.visit(Cypress.env("DEPLOYMENT_DRIVER_URL"), {
+          retryOnStatusCodeFailure: true,
+          retryOnNetworkFailure: true,
+          timeout: 5000,
+    })
+
+    //cy.get('#pf-login-username-id').type('admin') - hardcoded by Deployment Engine UI
+    // Enter the admin password for Deployment Engine UI login
+    cy.get('#pf-login-password-id').type(Cypress.env('DEPLOYMENT_ENGINE_UI_PASSWORD'))
+    cy.get('.pf-c-form__actions > .pf-c-button').click()
+
+    // Check the UI elements on the Red Hat login dialog screen
+    cy.get('.pf-c-modal-box__header > .pf-c-title')
+      .contains('A valid subscription for Ansible Automation Platform in your Red Hat account is required')
+      
+    // Check the messages on the Red Hat login dialog screen
+    cy.get('#pf-modal-part-2 > :nth-child(1)')
```

Comment:
Btw, just FYI, this way of implementing queries is very fragile, it will break as soon as the elements move within the DOM hierarchy. Better approach is to go by searching for displayed text and scoping searching to smaller area. We can adjust it later, just wanted to point it out.